### PR TITLE
[Lee Yong Ning] iP

### PR DESCRIPTION
**Pull request by yongning0310**
## Dong - Free Your Mind from Tasks!

> _“To Do or Not to Do, That Is the Question” – Unknown 

**Dong** helps you offload the burden of remembering every task. Why? Because:

- It's **text-based**.
- It's **easy to learn**.
- It's **FAST. SUPER FAST** to use.

### How to use?
1. **Download** it from [here](#).
2. **Double-click** to launch.
3. **Add** your tasks.
4. Let **Dong** manage them for you 😉.

### Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

📌 For all the Java programmers out there, you can use **Dong** to practice your Java skills. Check out the main method:

```java
    public static void main(String[] args) {
        new Duke("./data/tasks.txt").run();
    }
}

### CODE REVIEW

File: src/main/java/duke/Parser.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+        case FIND:
+            String findDescription = details[0].trim();
+            if (findDescription.length() <= 4) {
+                throw new InvalidFindException();
+            }
+            addToOutput(ui.printTaskContainingKeyword(taskList, findDescription.substring(5).trim()), output);
+            break;
+        default:
+            throw new InvalidInputException();
```
Comment (syamfarh) [inline src/main/java/duke/Parser.java:173]: I think you missed a break here

File: src/main/java/duke/Storage.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                    case "D":
+                        taskList.addDeadlineToList(isDone, taskParts[2].trim(), taskParts[3].trim());
+                        break;
+                    case "E":
+
+                        taskList.addEventToList(isDone, taskParts[2].trim(), taskParts[3].trim(), taskParts[4].trim());
+                        break;
+                    default:
+                        System.out.println("Unexpected task type encountered: " + line);
+                    }
```
Comment (syamfarh) [inline src/main/java/duke/Storage.java]: i think you missed a break here as well

File: src/main/java/duke/Storage.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+     */
+    public void writeToDB(TaskList taskList) {
+        StringBuilder content = new StringBuilder();
+        for (Task task : taskList.getList()) {
+            content.append(task.getDbString()).append("\n");
+        }
+
+        try {
+            Path path = Paths.get(this.filePath);
+            Files.write(path, content.toString().getBytes());
```
Comment (syamfarh) [inline src/main/java/duke/Storage.java]: does the writer object need to be closed after writing to file? I'm not sure if Files class need to be closed after used

File: src/main/java/duke/Ui.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public Ui(MainWindow mainWindow) {
+        this.mainWindow = mainWindow;
+        String logo = " ██▄   ████▄    ▄     ▄▀  \n"
+                + "█  █  █   █     █  ▄▀    \n"
+                + "█   █ █   █ ██   █ █ ▀▄  \n"
+                + "█  █  ▀████ █ █  █ █   █ \n"
+                + "███▀        █  █ █  ███  \n"
+                + "            █   ██       \n"
+                + "                         ";
```
Comment (syamfarh) [inline src/main/java/duke/Ui.java:21]: very nice logo!

**Review by syamfarh — COMMENTED** [review]
LGTM!

File: src/main/java/duke/DialogBox.java [hunk 5]
```diff
@@ -1,47 +1,61 @@
...
 import javafx.scene.control.Label;
+import javafx.scene.image.Image;
 import javafx.scene.image.ImageView;
 import javafx.scene.layout.HBox;
 
+/**
+ * An example of a custom control using FXML.
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
+ * containing text from the speaker.
+ */
```
Comment (miljyy) [inline src/main/java/duke/DialogBox.java]: I like how detailed and clear the Javadoc for this class is!

File: src/main/java/duke/taskclasses/TaskList.java [hunk 6]
```diff
@@ -76,13 +79,14 @@ public int getTasksCount() {
      * @param isDone The completion status of the task.
      * @param description Description of the task.
      */
-    public void addToDoToList(Boolean isDone, String description) {
+    public Task addToDoToList(Boolean isDone, String description) {
```
Comment (miljyy) [inline src/main/java/duke/taskclasses/TaskList.java:82]: Maybe the adding specific types of tasks could be handled by the subclasses of Task instead?

**Review by miljyy — COMMENTED** [review]
Overall the code is easy to read and understand, I really like how clean everything is!

**Pull request closed without merging**